### PR TITLE
Apply router pattern

### PR DESCRIPTION
**Pull request by AhmedAtef277**

### CODE REVIEW

File: e-store/e-store/Router/ViewControllerType.swift [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    func pop(animated: Bool = true) {
+        navigationController.popViewController(animated: animated)
+    }
+    
+    func popToRoot(animated: Bool = true) {
+        navigationController.popToRootViewController(animated: animated)
+    }
+    
+    func present(viewToPresent: ViewControllerType, animated: Bool = true, completion: (() -> Void)? = {}) {
+        navigationController.present(viewToPresent.navigationController, animated: animated, completion: completion)
```
Comment (Mahmoud-Bassuni) [inline e-store/e-store/Router/ViewControllerType.swift:38]: there is no UINavigationController on present

File: e-store/e-store/Router/ViewControllerType.swift [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    }
+    
+    func dismissView(animated: Bool = true, completion: (() -> Void)? = {}) {
+        navigationController.dismiss(animated: animated, completion: completion)
```
Comment (Mahmoud-Bassuni) [inline e-store/e-store/Router/ViewControllerType.swift:42]: the same

File: e-store/e-store/Router/StoreRouter.swift [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+//
+
+import Foundation
+
+protocol StoreRouterProtocol {
+    //TODO: - Navigation function here
+}
+
+class StoreRouter {
+    //TODO: - Implement navigation function here
```
Comment (Mahmoud-Bassuni) [inline e-store/e-store/Router/StoreRouter.swift:15]: where the implementation

Comment (AhmedAtef277) [reply e-store/e-store/Router/StoreRouter.swift:15]: who want to navigate will implement its navigation function

**Review by Mahmoud-Bassuni — APPROVED** [review]
thanks 🌹

**Pull request merged**